src/ring_theory/derivation.lean

```diff
@@ -122,6 +124,34 @@ instance : module A (derivation R A M) :=
 instance : is_scalar_tower R A (derivation R A M) :=
 ⟨λ x y z, ext (λ a, smul_assoc _ _ _)⟩
 
+section push_forward
+
+variables {N : Type*} [add_cancel_comm_monoid N] [module A N] [module R N] [is_scalar_tower R A N]
+variables (f : M →ₗ[A] N)
+
+/-- We can push forward derivations using linear maps, i.e., the composition of a derivation with a
+linear map is a derivation. Furthermore, this operation is linear on the spaces of derivations. -/
+def comp : derivation R A M →ₗ[R] derivation R A N :=
+{ to_fun    := λ D,
+  { leibniz'  := λ a b, by simp only [coe_fn_coe, function.comp_app, linear_map.coe_comp,
+      linear_map.coe_coe_is_scalar_tower, leibniz, linear_map.map_smul, linear_map.to_fun_eq_coe,
+      linear_map.map_add],
+    .. (f : M →ₗ[R] N).comp (D : A →ₗ[R] M), },
+  map_add'  := λ D₁ D₂, by ext; simp only [mk_coe, add_apply, coe_fn_coe, linear_map.coe_mk,
+    function.comp_app, linear_map.coe_comp, linear_map.to_fun_eq_coe, linear_map.map_add],
+  map_smul' := λ r D, by ext; simp only [coe_fn_coe, Rsmul_apply, linear_map.map_smul_of_tower,
+    mk_coe, linear_map.coe_mk, function.comp_app, linear_map.coe_comp, linear_map.to_fun_eq_coe], }
```

Comment:
Let's not use `by` unless it actually fits on one line (and not use `;` unless there are actually multiple goals), per the style guide! I think both of these things make proofs unnecessarily hard to human-parse.

Reply:
Agreed, thanks!